packages/cloud_firestore/cloud_firestore_web/lib/cloud_firestore_web.dart

```diff
@@ -57,7 +57,7 @@ class FirebaseFirestoreWeb extends FirebaseFirestorePlatform {
     try {
       await _webFirestore.clearPersistence();
     } catch (e) {
-      throw getFirebaseException(e);
```

Comment:
Considering how common this pattern is, it'd probably be worth to change it to:

```dart
Future<void> clearPersistence() async {
  return guard(() async {
    await _webFirestore.clearPersistence();
  });
}

R guard<R>(R Function() cb) {
  try {
    return cb();
  } catch (err, stack) {
    if (object is! core_interop.FirebaseError && object is Exception)
      rethrow;
    
    throw FirebaseException(...);
  }
}
```

Reply:
The `rethrow` is pretty important here too imo

Because if the goal is to support user exceptions, we want to preserve the stacktrace, or they'd have an hard time figuring out what the issue is.

Reply:
That is way better, didn't realise the stacktrace would be lost 👍